8-algoruthm-and-data-structures-2/hometask/src/JobRunner.ts

```diff
@@ -0,0 +1,79 @@
+interface Job {
+    priority: number;
+}
+
+class JobRunner {
+    private jobs: Job[];
+    
+    constructor(jobs: Job[]) {
+        this.jobs = jobs;
+    }
+
+    run(): void {
+        const sortedJobs = this.mergeSortJobByPriority(this.jobs);
+        this.doJobs(sortedJobs);
+    }
+
+    doJobs (jobs: Job[]){
+        let size = jobs.length;
+        while(size) {
+            const currentJob = jobs.pop();
+            if (currentJob){
+                size--;
+                console.log(`priority ${currentJob.priority}`);
+            }
+        }
+    }
+
+    merge(left: Job[], right: Job[]) {
```

Comment:
ideally all these merge/sort methods should be a part of separated class to manage elements with priority and that class can be injected to the job runner (DI can be used)